app/models/component.rb

```diff
@@ -16,27 +16,35 @@ class Component < Aggregate
              -> { where(usage: CONSTANTS::ENCRYPTION) }, class_name: 'Certificate', optional: true
 
   def self.to_service_metadata(event_id, published_at = Time.now)
-    matching_service_adapters = MsaComponent.includes(:enabled_signing_certificates,
-                                                      :encryption_certificate)
-    service_providers = SpComponent.includes(:enabled_signing_certificates,
-                                             :encryption_certificate)
+    matching_service_adapters = MsaComponent.all_components
+    service_providers = SpComponent.all_components
     components = matching_service_adapters + service_providers
 
     {
       published_at: published_at,
       event_id: event_id,
-      connected_services: components.map(&:services_to_metadata).flatten,
+      connected_services: components.map(&:services_to_metadata).flatten.compact,
```

Comment:
I was going to suggest adding `.select` here instead of the `if` below, when it was in its own the function, but it's fine now...